modules/java/generator/src/cpp/listconverters.cpp

```diff
@@ -0,0 +1,55 @@
+#define LOG_TAG "org.opencv.utils.Converters"
+#include "common.h"
+
+#define JAVA_UTIL_ARRAYLIST "java/util/ArrayList"
+
+
+jobject vector_String_to_List(JNIEnv* env, std::vector<cv::String>& vs) {
+
+    static jclass juArrayList       = static_cast<jclass>(env->NewGlobalRef(env->FindClass(JAVA_UTIL_ARRAYLIST)));
+    static jmethodID m_create   = env->GetMethodID(juArrayList, "<init>", "(I)V");
+    jmethodID m_add       = env->GetMethodID(juArrayList, "add", "(Ljava/lang/Object;)Z");
```

Comment:
I just took the example from [here](https://stackoverflow.com/questions/33406527/returning-an-arraylist-of-string-from-native-java-to-jni) As you can see only the class and constructor pointers are declared as static. Not sure if static is needed for other functions